Gems/ROS2/Code/Source/Lidar/LidarCore.cpp

```diff
@@ -161,8 +165,10 @@ namespace ROS2
         return m_lidarRaycasterId;
     }
 
-    RaycastResult LidarCore::PerformRaycast()
+    const RaycastResult& LidarCore::PerformRaycast()
```

Comment:
Any reason to change this to a reference other than to not need a copy constructor? Its a little risky to return a reference to a member variable because of its potential to change after this call is done by another function.

Reply:
The reason for this is to ommit copying of the raycast results which are quite large. It is highly unlikely that this reference will get invalidated since it is only used after each raycast. Until this reference is processed and discarded no further raycasts (for this object at least) will be scheduled.